lib/lure/vcs/vcs.go

```diff
@@ -0,0 +1,65 @@
+package vcs
+
+import (
+	"fmt"
+	"strings"
+)
+
+type header interface {
+	Add(key, value string)
+}
+
+type Authentication interface {
+	AuthenticateURL(url string) string
+	AuthenticateHTTPRequest(header header)
+}
+
+type TokenAuth struct {
+	Token string
+}
+
+func (auth TokenAuth) AuthenticateURL(url string) string {
+	return strings.Replace(url, "://", fmt.Sprintf("://x-token-auth:%s@", auth.Token), 1)
+}
+
+func (auth TokenAuth) AuthenticateHTTPRequest(header header) {
+	header.Add("Authorization", "Bearer "+auth.Token)
+}
+
+type UserPassAuth struct {
+	Username string
+	Password string
+}
+
+func (auth UserPassAuth) AuthenticateURL(url string) string {
+	return strings.Replace(url, "://", fmt.Sprintf("://%s:%s@", auth.Username, auth.Password), 1)
+}
+
+func (auth UserPassAuth) AuthenticateHTTPRequest(header header) {
+	//NOOP
+}
+
+type SourceControl interface {
```

Comment:
i'm a bit confused about this interface vs the other lower cased one

Reply:
In go, anything starting with an UpperCase is exported from the current package, lowerCase is not.